services/galley/src/Galley/API/Util.hs

```diff
@@ -48,22 +48,18 @@ ensureAccessRole role users mbTms = case role of
         when (length activated /= length users) $ throwM accessDenied
     NonActivatedAccessRole -> return ()
 
+-- | Check that the user is connected to everybody else.
+--
+-- The connection has to be bidirectional (e.g. if A connects to B and later
+-- B blocks A, the status of A-to-B is still 'Accepted' but it doesn't mean
+-- that they are connected).
 ensureConnected :: UserId -> [UserId] -> Galley ()
 ensureConnected _ []   = pure ()
 ensureConnected u uids = do
-    conns <- getConnections u uids (Just Accepted)
-    unless (all (isConnected u conns) uids) $
+    connsFrom <- getConnections [u] uids (Just Accepted)
+    connsTo   <- getConnections uids [u] (Just Accepted)
```

Comment:
I think we could/should do these 2 requests concurrently here (FWIW, the first request will be pretty fast, the 2nd might take longer) or even move this logic to brig